default/python/AI/savegame_codec/_decoder.py

```diff
@@ -43,14 +42,14 @@ def load_savegame_string(string):
     except zlib.error:
         pass  # probably an uncompressed (or wrongly base64 decompressed) string
     try:
-        decoded_state = decode(six.ensure_str(new_string, 'utf-8'))
+        decoded_state = decode(new_string)
```

Comment:
This should read ```decoded_state = decode(new_string.decode('utf-8'))```

Reply:
It is much more complex than that. We need to play with `str` and `bytes` in the chain in the case when we want to support old save formats.  

I will clean this in the separate PR.  

Do you know can we drop compatibility with old saves: not gzip and/or not based64?

Also, I would like to make change `def build_savegame_string(use_compression=True):` to `def build_savegame_string():` because it is used in that way for some times and this allows to simplify decoding.

PS. I found another strange thing: we send `bytes` to `fo.setSaveStateString` and get `str` in `resumeLoadedGame(saved_state_string)`

Reply:
> Do you know can we drop compatibility with old saves: not gzip and/or not based64?

By now it's probably fine to drop support for pre-base64 saves. They probably wouldn't load anyway due to other changes.

Also, presently even with XML compression disabled, the AI save state is still base-64 encoded, so isn't human readable anyway, so I assume that's not what you're talking about dropping support for.

Reply:
> Do you know can we drop compatibility with old saves: not gzip and/or not based64?

Fine with me. It's trivial to reimplement the old behavior if necessary anyway.